### PR TITLE
Graph layout is now saved when navigate away from page

### PR DESCRIPTION
**Pull request by p013570**

### CODE REVIEW

File: ui/src/main/webapp/app/graph/graph-component.js [hunk 1]
```diff
@@ -630,9 +654,24 @@ function GraphController($q, graph, config, error, loading, query, operationOpti
      * Removes every selected element in the graph.
      */
     vm.removeSelected = function() {
-        cytoscapeGraph.filter(":selected").remove();
+        var removedElements = cytoscapeGraph.filter(":selected").remove();
+        for(var i = 0; i < removedElements.length; i++) {
```
Comment (d47853) [inline ui/src/main/webapp/app/graph/graph-component.js]: could you use array.concat() here? https://devdocs.io/javascript/global_objects/array/concat

File: ui/src/main/webapp/app/graph/graph-component.js [hunk 2]
```diff
@@ -630,9 +654,24 @@ function GraphController($q, graph, config, error, loading, query, operationOpti
...
+        }
+
         cytoscapeGraph.elements().unselect();
         vm.selectedElements.entities = [];
         vm.selectedElements.edges = [];
     }
+
+    var removePreviouslyRemovedElements = function() {
+        var elements = cytoscapeGraph.elements();
+        for(var i = 0; i < graph.getRemovedElements().length; i++) {
```
Comment (d47853) [inline ui/src/main/webapp/app/graph/graph-component.js]: Could you use a for each loop here? https://devdocs.io/javascript/statements/for...of

File: ui/src/main/webapp/app/graph/graph-component.js (hunk 1 above)
Comment (p013570) [reply ui/src/main/webapp/app/graph/graph-component.js]: Unfortunately removedElements is not actually an array, so I can't use that.

File: ui/src/main/webapp/app/graph/graph-component.js (hunk 2 above)
Comment (p013570) [reply ui/src/main/webapp/app/graph/graph-component.js]: Unfortunately elements is a funny object and if I try and use that I get this error:
```Uncaught TypeError: elements is not iterable```

File: ui/src/main/webapp/app/graph/graph-component.js (hunk 1 above)
Comment (d47853) [reply ui/src/main/webapp/app/graph/graph-component.js]: You could use the [toArray method](https://js.cytoscape.org/#eles.toArray)

File: ui/src/main/webapp/app/graph/graph-component.js (hunk 2 above)
Comment (d47853) [reply ui/src/main/webapp/app/graph/graph-component.js]: You could use the [eles.forEach method](https://js.cytoscape.org/#eles.forEach)

**Review by d47853 — DISMISSED** [review]

**Review by t11947 — APPROVED** [review]

**Comment by n3101** [comment]
@t11947 Hi, am I right in thinking the only thing blocking this PR is the failure of the selenium tests? If so, @GCHQDeveloper314 has started a rumour that a selenium version upgrade MAY have made that problem go away. Would you be able to try it again and see if that is true?

**Pull request merged**